### PR TITLE
k0s-in-docker

### LINKED ISSUE #282: MKE in docker for local dev
We want to enable easy setup of non-production MKE in Docker.

## Step 1.

As a user I want to be able to easily spin up local single node MKE:
```
docker run -t -i --rm -p 6443:6443 --privileged -v /var/lib/mke --name mke --hostname mke-node1 mke
```

## Step 2.

Make it possible to spin up multi-node cluster. Could be done with docker-compose and some clever entrypoint scripting. If e.g. the controller container also dumps a token to a shared volume, the worker containers can then wait until that file exists and then join the cluster.

### PR DESCRIPTION
**Pull request by jnummelin**
Signed-off-by: Jussi Nummelin <jnummelin@mirantis.com>

**Issue**
Fixes #282 

**What this PR Includes**

Basic setup for building k0s into container.

**Testing:**

Tested locally with:
```
docker build -t docker.pkg.github.com/k0sproject/k0s/k0s:latest .
```
```
docker run -d --name k0s --hostname k0s --privileged -v /var/lib/k0s -p 6443:6443 docker.pkg.github.com/k0sproject/k0s/k0s:latest
```
```
token=$(docker exec -t -i k0s k0s token create --role=worker)
```
```
docker run -d --name k0s-worker1 --hostname k0s-worker1 --privileged -v /var/lib/k0s docker.pkg.github.com/k0sproject/k0s/k0s:latest k0s worker $token
```
```
$ docker exec -t -i k0s sh
/ # kubectl get node
NAME          STATUS   ROLES    AGE     VERSION
k0s           Ready    <none>   9m26s   v1.19.3
k0s-worker1   Ready    <none>   9m26s   v1.19.3
```

### CODE REVIEW

File: docs/k0s-in-docker.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+```
+Just grab the kubeconfig with `docker exec k0s-controller cat /var/lib/k0s/pki/admin.conf` and paste e.g. into [Lens](https://github.com/lensapp/lens/).
+
+## Running workers
+
+If you want to attach multiple workers nodes into the cluster you can run separate containers for workers.
+
+First, we need a join token for the worker:
+```sh
+token=$(docker exec -t -i k0s k0s token create --role=worker)
```
Comment (jasmingacic) [inline docs/k0s-in-docker.md:17]: Command returns this error:
```
token=`docker exec 568503a4fb53 k0s token create --role-worker` 
2020-11-05 09:09:59.142734 I | flag provided but not defined: -role-worker
```

File: docs/k0s-in-docker.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# k0s in Docker
+
+We publish k0s container image for every release. By default we run both controller and worker in the same container to allow easy local test "cluster".
+
+You can run your own k0s-in-docker easily with:
+```sh
+docker run -d --name k0s --hostname k0s --privileged -v /var/lib/k0s -p 6443:6443 docker.pkg.github.com/k0sproject/k0s/k0s:<version>
```
Comment (jasmingacic) [inline docs/k0s-in-docker.md:7]: The image is not publicly available:
```
$ docker run -d --name k0s --hostname k0s --privileged -v /var/lib/k0s -p 6443:6443 docker.pkg.github.com/k0sproject/k0s/k0s:latest
Unable to find image 'docker.pkg.github.com/k0sproject/k0s/k0s:latest' locally
docker: Error response from daemon: Get https://docker.pkg.github.com/v2/k0sproject/k0s/k0s/manifests/latest: no basic auth credentials.
```

Comment (jnummelin) [reply docs/k0s-in-docker.md:7]: I know, but it will be after the next release

File: docs/k0s-in-docker.md (hunk 1 above)
Comment (jnummelin) [reply docs/k0s-in-docker.md:17]: You've got a typo, `--role-worker` vs. `--role=worker`

Comment (jasmingacic) [reply docs/k0s-in-docker.md:17]: I just realised that :D

**Review by soider — APPROVED** [review]

**Review by jasmingacic — APPROVED** [review]

**Pull request merged**